### PR TITLE
Add nic feature discovery

### PR DESCRIPTION
**Pull request by adrianchiris**

### CODE REVIEW

**Review by rollandf — APPROVED** [review]
LGTM

**Comment by adrianchiris** [comment]
@e0ne @ykulazhenkov  PTAL, id like to try and merge this one by EOW.

File: manifests/state-nic-feature-discovery/030-nic-feature-discovery-ds.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        tier: node
+        app: nic-feature-discovery
+        name: nic-feature-discovery
+    spec:
+      terminationGracePeriodSeconds: 10
+      {{ if eq .RuntimeSpec.OSName "rhcos" }}
+      serviceAccountName: nic-feature-discovery
+      {{- end }}
+      tolerations:
+      - key: nvidia.com/gpu
```
Comment (ykulazhenkov) [inline manifests/state-nic-feature-discovery/030-nic-feature-discovery-ds.yaml:42]: It looks like we are missing support for custom tollerations here
```
        {{- if .Tolerations }}
        {{- .Tolerations | yaml | nindent 8 }}
        {{- end }}
```

Comment (adrianchiris) [reply manifests/state-nic-feature-discovery/030-nic-feature-discovery-ds.yaml]: fixed !

**Review by ykulazhenkov — APPROVED** [review]

File: pkg/state/factory.go [hunk 2]
```diff
@@ -121,12 +121,18 @@ func newNicClusterPolicyStates(k8sAPIClient client.Client, scheme *runtime.Schem
 	nvIpamCniState, err := NewStateNVIPAMCNI(
 		k8sAPIClient, scheme, filepath.Join(manifestBaseDir, "state-nv-ipam-cni"))
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to create ib-kubernetes State")
```
Comment (e0ne) [inline pkg/state/factory.go:124]: thanks for the fix!

**Review by e0ne — APPROVED** [review]
LGTM

File: pkg/state/factory.go (hunk 2 above)
Comment (adrianchiris) [reply pkg/state/factory.go:124]: np, while not related to this PR, it just bothered me :) plus im the one who originally added it.

**Comment by adrianchiris** [comment]
@abdallahyas shouldnt we have CI reporting ?

**Pull request merged**